### DISCUSSION TITLE
Wi-Fi conncetion, NIC detected but I think I am missing something

### DISCUSSION THREAD

**Discussion by NerfyGek0** [ipxe/ipxe#1361] [Q&A]
Hi,

I would like to get ipxe to connect to my Wi-Fi network using a bootable USB. before I start writing scripts, flashing firmware etc. 
It all seems to go quite well except it does not actually connect to my network. 

I am trying to keep it simple:
- I have a Atheros 9300 PCI-E [168c:0030] that is listed as a compatible device
- Using the default ipxe efi with the ath9k added explicitly (just to make sure driver is available)
- burning this ISO to a USB and booting off this
- Have tried to connect to home network with WPA2, Phone hotspot WPA2 and hotspot with no encryption.

The wireless nic shows up in ipxe, using config I enter in the SSID as my Wi-Fi name and the key my wifi password but I get the "Connection Timed out" message. 

Steps:
1. boot USB stick
2. Ctrl-B to enter shell
3. iwstat < shows my Wi-Fi card on net0
4. config 
5. Enter SSID and key on net0
6. exit config
7. ifopen net0
8. iwstat (see screenshot)
9. If I run iwstat multiple times it says "802.11 not associated" each time but I can see the Chanel number changing. 

I enabled DEBUG=net80211 and got the following output (sorry for the glare but its just the last two red lines repeated 3 times): 
![debug](https://github.com/user-attachments/assets/f917a046-a375-4067-83b4-329e60b988ec)

Questions: 

- Do I need to specify WPA2 anywhere or just it just need to be enabled in the config/general.h?
- Are all 802.11 standards supported provided the NIC supports them? Do I need to explicitly enable them?
- Anything I am missing here?
- Does the debug output from the screenshot tell you anything about my issue?

Thanks for any help on this, would love to have this work; just for fun to see if it can. :)

**Comment by NiKiZe** [comment]
Try first without encryption. Since you have enabled `iwstat` command that should pull in everything else.

WPA2 might not be supported.

**Reply by NerfyGek0** [reply]
These lines are in the general.h code so I assume WPA2 is supported. But yes, I tried a hot spotting my phone without any encryption.

`#define	CRYPTO_80211_WEP	/* WEP encryption (deprecated and insecure!) */

#define	CRYPTO_80211_WPA	/* WPA Personal, authenticating with passphrase */

#define	CRYPTO_80211_WPA2	/* Add support for stronger WPA cryptography */`
